Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67117
* __->__ #66904

Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang

zhaojuanmao committed:
Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper

Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

[ghstack-poisoned]

mrshenli reviewed (commented):
mypy test failure is real:

```
torch/testing/_internal/common_fsdp.py:202:13: error: "Parameter" has no attribute "expert"  [attr-defined]
torch/testing/_internal/common_fsdp.py:214:22: error: Incompatible types in assignment (expression has type "FullyShardedDataParallel", variable has type "Linear")  [assignment]
torch/testing/_internal/common_fsdp.py:216:22: error: Incompatible types in assignment (expression has type "FullyShardedDataParallel", variable has type "Linear")  [assignment]
Found 3 errors in 1 file (checked 1512 source files)
```

zhaojuanmao committed:
Update on "Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper"


Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

zhaojuanmao committed:
Update on "Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper"


Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

zhaojuanmao committed:
Update on "Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper"


Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

mrshenli reviewed (commented):
BTW, I assume this adds all test files in the fsdp folder to CI?

https://github.com/pytorch/pytorch/blob/23321ba7a3b634ee734455aab4a984689802cad0/test/run_test.py#L137

zhaojuanmao committed:
Update on "Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper"


Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

mrshenli reviewed (approved):
LGTM! Thanks!

zhaojuanmao committed:
Update on "Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper"


Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]

zhaojuanmao committed:
Update on "Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrapper"


Add more FSDP unit tests to cover core logic, freezing weights and flatten parameter wrappe, these unit tests are refactored to be aligned with PyTorch commonly used test classes

Differential Revision: [D31779565](https://our.internmc.facebook.com/intern/diff/D31779565/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31779565/)!

cc pietern mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse @SciPioneer @H-Huang

[ghstack-poisoned]
